### PR TITLE
helm: Allow configure the scrape interval for ServiceMonitor

### LINKED ISSUE #19978: CFP: [Helm chart] Allow configuration of serviceMonitor interval
## Cilium Feature Proposal

**Describe the feature you'd like**

I want to configure the scrape interval for the helm chart's included ServiceMonitor, I want to raise the seconds a bit to lower my costs.
This affects the operator, agent and hubble deployments

**(Optional) Describe your proposed solution**

Allow an extra option "interval" for the serviceMonitor blocks in `values.yaml` as such:
```
serviceMonitor:
  enabled: false
  labels: {}
  annotations: {}
  interval: "10s"
```

### PR DESCRIPTION
**Pull request by LawlietLi**
This commit added a configuration item for serviceMonitor, Allow configure the scrape interval for ServiceMonitor
Fixes: #19978

Signed-off-by: Li Yiheng <lyhutopi@gmail.com>

Please ensure your pull request adheres to the following guidelines:

- [x] For first time contributors, read [Submitting a pull request](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#submitting-a-pull-request)
- [x] All code is covered by unit and/or runtime tests where feasible.
- [x] All commits contain a well written commit description including a title,
      description and a `Fixes: #XXX` line if the commit addresses a particular
      GitHub issue.
- [x] All commits are signed off. See the section [Developer’s Certificate of Origin](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#dev-coo)
- [x] Provide a title or release-note blurb suitable for the release notes.
- [x] Thanks for contributing!

### CODE REVIEW

File: install/kubernetes/cilium/templates/cilium-agent/servicemonitor.yaml [hunk 1]
```diff
@@ -22,7 +22,7 @@ spec:
     - {{ .Release.Namespace }}
   endpoints:
   - port: metrics
-    interval: 10s
+    interval: {{ .Values.prometheus.serviceMonitor.interval }}
```
Comment (kaworu) [inline install/kubernetes/cilium/templates/cilium-agent/servicemonitor.yaml]:
```suggestion
    interval: {{ .Values.prometheus.serviceMonitor.interval | quote }}
```

File: install/kubernetes/cilium/templates/cilium-operator/servicemonitor.yaml [hunk 2]
```diff
@@ -22,7 +22,7 @@ spec:
     - {{ .Release.Namespace }}
   endpoints:
   - port: metrics
-    interval: 10s
+    interval: {{ .Values.operator.prometheus.serviceMonitor.interval }}
```
Comment (kaworu) [inline install/kubernetes/cilium/templates/cilium-operator/servicemonitor.yaml]:
```suggestion
    interval: {{ .Values.operator.prometheus.serviceMonitor.interval | quote }}
```

File: install/kubernetes/cilium/templates/hubble/servicemonitor.yaml [hunk 3]
```diff
@@ -21,7 +21,7 @@ spec:
     - {{ .Release.Namespace }}
   endpoints:
   - port: hubble-metrics
-    interval: 10s
+    interval: {{ .Values.hubble.metrics.serviceMonitor.interval }}
```
Comment (kaworu) [inline install/kubernetes/cilium/templates/hubble/servicemonitor.yaml]:
```suggestion
    interval: {{ .Values.hubble.metrics.serviceMonitor.interval | quote }}
```

**Review by kaworu — CHANGES_REQUESTED** [review]
Thanks for the PR @LawlietLi! I have a minor nitpicking about quoting the interval values, but other than that LGTM :+1:

**Comment by LawlietLi** [comment]
> Thanks for the PR @LawlietLi! I have a minor nitpicking about quoting the interval values, but other than that LGTM 👍

Thanks, I've modified this.

**Review by kaworu — CHANGES_REQUESTED** [review]
Sorry @LawlietLi, we've recently made some changes around `values.yaml` and it is now generated from a "source" `values.yaml.tmpl` file (see [this PR](https://github.com/cilium/cilium/pull/20066) for more context). Could you please:

1. rebase your PR against the `master` branch
2. port the PR's changes from `install/kubernetes/values.yaml` to `install/kubernetes/values.yaml.tmpl`
3. run `make -C install/kubernetes`

Thank you in advance :pray:

**Comment by LawlietLi** [comment]
I have rebased to master and regenerated `values.yaml`.

**Review by tommyp1ckles — APPROVED** [review]
lgtm, thanks!

**Review by kaworu — APPROVED** [review]
Thanks @LawlietLi!

**Review by kaworu — CHANGES_REQUESTED** [review]
[Travis-CI failure](https://app.travis-ci.com/github/cilium/cilium/jobs/574089637) is legit:

> HINT: to fix this, run 'make -C Documentation update-helm-values'

**Review by kaworu — COMMENTED** [review]
CI is passing (no need for a full run), @LawlietLi can you squash all commits into the first one please?

**Comment by LawlietLi** [comment]
OK, I will squash commits.

**Review by kaworu — APPROVED** [review]

**Comment by kaworu** [comment]
No need for a full CI run here, marking as `ready-to-merge`.

**Pull request merged**